### ISSUE TITLE
FatJar appears broken for 2.4.0RC1

### ISSUE THREAD

**Issue by bearrito** [OpenTSDB/opentsdb#1027]
```
  opentsdb git:(v2.4.0RC1) mvn -f fat-jar-pom.xml -Dgpg.skip -DskipTests clean install
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building OpenTSDB 2.4.0RC1
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ opentsdb-fatjar ---
[INFO] Deleting /tmp/opentsdb/target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-maven) @ opentsdb-fatjar ---
[INFO] 
[INFO] --- exec-maven-plugin:1.2.1:exec (generate-build-data) @ opentsdb-fatjar ---
Generating target/generated-sources/net/opentsdb/tools/BuildData.java
[INFO] 
[INFO] --- exec-maven-plugin:1.2.1:exec (build-data-codegen) @ opentsdb-fatjar ---
Generating target/generated-sources/net/opentsdb/BuildData.java
[INFO] 
[INFO] --- build-helper-maven-plugin:1.7:add-source (add-source) @ opentsdb-fatjar ---
[INFO] Source directory: /tmp/opentsdb/target/generated-sources added.
[INFO] 
[INFO] --- ph-javacc-maven-plugin:2.8.0:javacc (jjc) @ opentsdb-fatjar ---
Java Compiler Compiler Version 6.1_2 (Parser Generator)
(type "javacc" with no arguments for help)
Reading from file /tmp/opentsdb/src/parser.jj . . .
Warning: Lookahead adequacy checking not being performed since option LOOKAHEAD is more than 1.  Set option FORCE_LA_CHECK to true to force checking.
File "TokenMgrError.java" does not exist.  Will create one.
File "ParseException.java" does not exist.  Will create one.
File "Token.java" does not exist.  Will create one.
File "SimpleCharStream.java" does not exist.  Will create one.
Parser generated with 0 errors and 1 warnings.
[INFO] Processed 1 grammar
[INFO] 
[INFO] --- maven-resources-plugin:2.3:resources (default-resources) @ opentsdb-fatjar ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-antrun-plugin:1.7:run (default) @ opentsdb-fatjar ---
[INFO] Executing tasks

main:
     [copy] Copying 1 file to /tmp/opentsdb/target/classes
     [copy] Copying 1 file to /tmp/opentsdb/target/classes
[INFO] Executed tasks
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ opentsdb-fatjar ---
[INFO] Compiling 218 source files to /tmp/opentsdb/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[18,32] error: package com.esotericsoftware.kryo does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[19,32] error: package com.esotericsoftware.kryo does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[20,35] error: package com.esotericsoftware.kryo.io does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[21,35] error: package com.esotericsoftware.kryo.io does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[83,42] error: cannot find symbol
[ERROR]   symbol:   class KryoSerializable
  location: interface HistogramDataPoint
/tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[167,22] error: cannot find symbol
[ERROR]   symbol:   class Kryo
  location: class HistogramBucket
/tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[167,33] error: cannot find symbol
[ERROR]   symbol:   class Output
  location: class HistogramBucket
/tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[172,21] error: cannot find symbol
[ERROR]   symbol:   class Kryo
  location: class HistogramBucket
/tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[172,32] error: cannot find symbol
[ERROR]   symbol:   class Input
  location: class HistogramBucket
/tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[28,32] error: package com.esotericsoftware.kryo does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[29,35] error: package com.esotericsoftware.kryo.io does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[30,35] error: package com.esotericsoftware.kryo.io does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[216,20] error: cannot find symbol
[ERROR]   symbol:   class Kryo
  location: class SimpleHistogram
/tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[216,31] error: cannot find symbol
[ERROR]   symbol:   class Output
  location: class SimpleHistogram
/tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[229,19] error: cannot find symbol
[ERROR]   symbol:   class Kryo
  location: class SimpleHistogram
/tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[229,30] error: cannot find symbol
[ERROR]   symbol:   class Input
  location: class SimpleHistogram
/tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[30,45] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[30,45] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/TsdbQuery.java:[571,23] [deprecation] DownsamplingSpecification(long,Aggregator,FillPolicy) in DownsamplingSpecification has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/TsdbQuery.java:[1316,28] [deprecation] DownsamplingSpecification(long,Aggregator,FillPolicy) in DownsamplingSpecification has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/TsdbQuery.java:[1675,20] [deprecation] DownsamplingSpecification(long,Aggregator,FillPolicy) in DownsamplingSpecification has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/utils/ByteSet.java:[83,47] [unchecked] unchecked cast
[ERROR]   required: ByteMap<Object>
  found:    Object
/tmp/opentsdb/src-main/net/opentsdb/core/Span.java:[500,13] [deprecation] Downsampler(SeekableView,long,Aggregator) in Downsampler has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/Span.java:[504,13] [deprecation] FillingDownsampler(SeekableView,long,long,long,Aggregator,FillPolicy) in FillingDownsampler has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[129,15] [deprecation] getHeaders() in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[130,53] [deprecation] getHeaders() in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[155,15] [deprecation] getHeaders() in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[156,53] [deprecation] getHeaders() in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[460,18] [deprecation] toStringHelper(Object) in Objects has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[480,15] [deprecation] containsHeader(String) in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[482,35] [deprecation] getHeader(String) in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/QueryRpc.java:[471,34] [deprecation] getLastPoint(TSDB,byte[],boolean,int,long) in TSUIDQuery has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/utils/JSON.java:[214,36] [deprecation] createJsonParser(String) in JsonFactory has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/utils/JSON.java:[236,36] [deprecation] createJsonParser(byte[]) in JsonFactory has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/utils/JSON.java:[258,36] [deprecation] createJsonParser(InputStream) in JsonFactory has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SpanGroup.java:[198,13] [deprecation] DownsamplingSpecification(long,Aggregator,FillPolicy) in DownsamplingSpecification has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[74,4] error: cannot find symbol
[ERROR]   symbol:   class Output
  location: class SimpleHistogram
/tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[74,24] error: cannot find symbol
[ERROR]   symbol:   class Output
  location: class SimpleHistogram
/tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[102,4] error: cannot find symbol
[ERROR]   symbol:   class Input
  location: class SimpleHistogram
/tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[104,18] error: cannot find symbol
[ERROR]   symbol:   class Input
  location: class SimpleHistogram
/tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[250,77] [unchecked] unchecked cast
[ERROR]   required: Set<Entry<HistogramBucket,Long>>
  found:    Set
/tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[448,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[453,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[458,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[463,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[483,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[488,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[493,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[341,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[360,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[365,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[370,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[390,14] NotImplementedException is internal proprietary API and may be removed in a future release
[INFO] 53 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 6.087s
[INFO] Finished at: Tue Jul 18 12:03:50 EDT 2017
[INFO] Final Memory: 27M/397M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project opentsdb-fatjar: Compilation failure: Compilation failure:
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[18,32] error: package com.esotericsoftware.kryo does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[19,32] error: package com.esotericsoftware.kryo does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[20,35] error: package com.esotericsoftware.kryo.io does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[21,35] error: package com.esotericsoftware.kryo.io does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[83,42] error: cannot find symbol
[ERROR] symbol:   class KryoSerializable
[ERROR] location: interface HistogramDataPoint
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[167,22] error: cannot find symbol
[ERROR] symbol:   class Kryo
[ERROR] location: class HistogramBucket
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[167,33] error: cannot find symbol
[ERROR] symbol:   class Output
[ERROR] location: class HistogramBucket
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[172,21] error: cannot find symbol
[ERROR] symbol:   class Kryo
[ERROR] location: class HistogramBucket
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/HistogramDataPoint.java:[172,32] error: cannot find symbol
[ERROR] symbol:   class Input
[ERROR] location: class HistogramBucket
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[28,32] error: package com.esotericsoftware.kryo does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[29,35] error: package com.esotericsoftware.kryo.io does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[30,35] error: package com.esotericsoftware.kryo.io does not exist
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[216,20] error: cannot find symbol
[ERROR] symbol:   class Kryo
[ERROR] location: class SimpleHistogram
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[216,31] error: cannot find symbol
[ERROR] symbol:   class Output
[ERROR] location: class SimpleHistogram
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[229,19] error: cannot find symbol
[ERROR] symbol:   class Kryo
[ERROR] location: class SimpleHistogram
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[229,30] error: cannot find symbol
[ERROR] symbol:   class Input
[ERROR] location: class SimpleHistogram
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[30,45] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[30,45] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/TsdbQuery.java:[571,23] [deprecation] DownsamplingSpecification(long,Aggregator,FillPolicy) in DownsamplingSpecification has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/TsdbQuery.java:[1316,28] [deprecation] DownsamplingSpecification(long,Aggregator,FillPolicy) in DownsamplingSpecification has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/TsdbQuery.java:[1675,20] [deprecation] DownsamplingSpecification(long,Aggregator,FillPolicy) in DownsamplingSpecification has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/utils/ByteSet.java:[83,47] [unchecked] unchecked cast
[ERROR] required: ByteMap<Object>
[ERROR] found:    Object
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/Span.java:[500,13] [deprecation] Downsampler(SeekableView,long,Aggregator) in Downsampler has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/Span.java:[504,13] [deprecation] FillingDownsampler(SeekableView,long,long,long,Aggregator,FillPolicy) in FillingDownsampler has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[129,15] [deprecation] getHeaders() in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[130,53] [deprecation] getHeaders() in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[155,15] [deprecation] getHeaders() in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[156,53] [deprecation] getHeaders() in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[460,18] [deprecation] toStringHelper(Object) in Objects has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[480,15] [deprecation] containsHeader(String) in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/AbstractHttpQuery.java:[482,35] [deprecation] getHeader(String) in HttpMessage has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/tsd/QueryRpc.java:[471,34] [deprecation] getLastPoint(TSDB,byte[],boolean,int,long) in TSUIDQuery has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/utils/JSON.java:[214,36] [deprecation] createJsonParser(String) in JsonFactory has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/utils/JSON.java:[236,36] [deprecation] createJsonParser(byte[]) in JsonFactory has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/utils/JSON.java:[258,36] [deprecation] createJsonParser(InputStream) in JsonFactory has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SpanGroup.java:[198,13] [deprecation] DownsamplingSpecification(long,Aggregator,FillPolicy) in DownsamplingSpecification has been deprecated
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[74,4] error: cannot find symbol
[ERROR] symbol:   class Output
[ERROR] location: class SimpleHistogram
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[74,24] error: cannot find symbol
[ERROR] symbol:   class Output
[ERROR] location: class SimpleHistogram
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[102,4] error: cannot find symbol
[ERROR] symbol:   class Input
[ERROR] location: class SimpleHistogram
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[104,18] error: cannot find symbol
[ERROR] symbol:   class Input
[ERROR] location: class SimpleHistogram
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/core/SimpleHistogram.java:[250,77] [unchecked] unchecked cast
[ERROR] required: Set<Entry<HistogramBucket,Long>>
[ERROR] found:    Set
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[448,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[453,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[458,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[463,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[483,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[488,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/IntersectionIterator.java:[493,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[341,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[360,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[365,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[370,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] /tmp/opentsdb/src-main/net/opentsdb/query/expression/UnionIterator.java:[390,14] NotImplementedException is internal proprietary API and may be removed in a future release
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

**Comment by nickman** [comment]
I'll take this one. Looks like the fat-jar pom template needs updating for new dependencies.

**Comment by bearrito** [comment]
All I had to do was add 

```
<dependency>
      <groupId>com.esotericsoftware.kryo</groupId>
      <artifactId>kryo</artifactId>
      <version>2.21.1</version>
</dependency>
```

and do the modification required for building on java 8

```
[INFO] Installing /tmp/opentsdb/target/opentsdb-fatjar-2.4.0RC1.jar to /home/barrett/.m2/repository/net/opentsdb/opentsdb-fatjar/2.4.0RC1/opentsdb-fatjar-2.4.0RC1.jar
[INFO] Installing /tmp/opentsdb/fat-jar-pom.xml to /home/barrett/.m2/repository/net/opentsdb/opentsdb-fatjar/2.4.0RC1/opentsdb-fatjar-2.4.0RC1.pom
[INFO] Installing /tmp/opentsdb/target/opentsdb-fatjar-2.4.0RC1-sources.jar to /home/barrett/.m2/repository/net/opentsdb/opentsdb-fatjar/2.4.0RC1/opentsdb-fatjar-2.4.0RC1-sources.jar
[INFO] Installing /tmp/opentsdb/target/opentsdb-fatjar-2.4.0RC1-javadoc.jar to /home/barrett/.m2/repository/net/opentsdb/opentsdb-fatjar/2.4.0RC1/opentsdb-fatjar-2.4.0RC1-javadoc.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1:03.854s
[INFO] Finished at: Tue Jul 18 12:14:34 EDT 2017
[INFO] Final Memory: 91M/1049M
[INFO] ------------------------------------------------------------------------


```

**Comment by nickman** [comment]
Super. Does the fat-jar work ok ?

**Comment by bearrito** [comment]
I'm am about to fire it up it a few moments. I'll report back with results.

**Comment by bearrito** [comment]
I ran a simple integration test against my local hbase instance without issues. 
I didn't run any command using the streaming quantile functionality. I might try that later after I digest the api after review of testcode

**Comment by manolama** [comment]
Fixed in a89d1847831d34d4edd3b6fa059ad60c5e29020f, thanks!
